dashboard/app/controllers/pd/workshop_daily_survey_controller.rb

```diff
@@ -225,6 +225,67 @@ def new_post
       render :new_general
     end
 
+    # GET /pd/workshop_survey/csf/:survey_name/:enrollment_code
+    def new_csf
+      params[:survey_name] = params[:survey_name]&.downcase
+      return render_404 unless CSF_SURVEY_INDEXES.key? params[:survey_name]
+
+      if params[:survey_name] == PRE_DEEPDIVE_SURVEY
+        # TODO: what if we don't require log-in? what if pre-ws survey can be annonymous
```

Comment:
```suggestion
        # TODO: what if we don't require log-in? what if pre-ws survey can be anonymous?
```
Is this TODO something you meant to address now, or are you intentionally leaving it for future follow-up?

Reply:
I'd be tempted to say we should respond with a 404 if someone tries to pass an enrollment code to the pre-survey, since that's not a combination we currently support.

Alternatively, a lot of this controller method is this large if/else statement with some common code at the end. I think that's an antipattern.  It might be clearer to have separate controller methods, and extract a helper for the common code they both use.

Put another way, should we turn this:

```rb
# GET /pd/workshop_survey/csf/:survey_name/:enrollment_code
def new_csf
  # Validation of survey_name parameter
  if params[:survey_name] == PRE_DEEPDIVE_SURVEY
    # pre-deepdive-survey logic
  elsif params[:survey_name] == POST_DEEPDIVE_SURVEY
    # post-deepdive-survey logic
  end
  # common render logic
end
```

into this:

```rb
# GET /pd/workshop_survey/csf/pre201
def new_pre201
  # pre-deepdive-survey logic
  render_survey(@form_id, workshop)
end

# GET /pd/workshop_survey/csf/post201(/:enrollment_code)
def new_post201
  # post-deepdive-survey logic
  render_survey(@form_id, workshop)
end

def render_survey
  # common render logic
end
```

With less branching and validation logic, it may be easier to reason about in the future without much duplication of code.

Reply:
I like your suggestion, it looks cleaner, will fix it.
About the TODO, it just me thinking out loud (in writing). I forgot to delete it before committing.